www/data_module/src/services/data/collection/dataquery.service.coffee

```diff
@@ -0,0 +1,77 @@
+class DataQuery extends Factory
+    constructor: ($http, $q, API) ->
+        return class DataQueryClass
+            constructor: (query = {}) ->
+                @query = query
+                @filters = {}
+                for fieldAndOperator, value of query
+                    if ['field', 'limit', 'offset', 'order'].indexOf(fieldAndOperator) < 0
+                        if ['on', 'true', 'yes'].indexOf(value) > -1 then value = true
+                        else if ['off', 'false', 'no'].indexOf(value) > -1 then value = false
+                        @filters[fieldAndOperator] = value
+
+            computeQuery: (array) ->
+                # 1. filtering
+                @filter(array)
+
+                # 2. sorting
+                order = @query?.order
+                @sort(array, order)
+
+                # 3. limit
+                limit = @query?.limit
+                @limit(array, limit)
+
+
+            isFiltered: (v) ->
+                cmp = false
+                for fieldAndOperator, value of @filters
+                    [field, operator] = fieldAndOperator.split('__')
+                    switch operator
+                        when 'ne' then cmp = v[field] != value
+                        when 'lt' then cmp = v[field] <  value
+                        when 'le' then cmp = v[field] <= value
+                        when 'gt' then cmp = v[field] >  value
+                        when 'ge' then cmp = v[field] >= value
+                        else cmp = v[field] == value or
+                            (angular.isArray(v[field]) and value in v[field]) or
+                            # private fields added by the data service
+                            v["_#{field}"] == value or
+                            (angular.isArray(v["_#{field}"]) and value in v["_#{field}"])
```

Comment:
A make a pull request here https://github.com/tardyp/buildbot/pull/7 
Though the list of builds is displaying now, we still have problems displaying a single build. I will look into that later.